src/GitTfsTest/Integration/CloneTests.cs

```diff
@@ -235,6 +235,39 @@ public void WhenCloningFunctionalTestVtccdsWithBranchesRenaming_ThenAllRenamesSh
             h.AssertNoRef("Vtccds", "refs/remotes/tfs/testRename");
         }
 
+        [FactExceptOnUnix]
+        public void WhenCloningFunctionalTestVtccdsWithBranchesRenamingAndGitignore_ThenAllRenamesShouldBeWellHandled()
+        {
+            string gitignoreFile = Path.Combine(h.Workdir, "gitignore");
+            string gitignoreContent = "*.exe\r\n*.com\r\n";
+            File.WriteAllText(gitignoreFile, gitignoreContent);
+
+            h.SetupFake(r =>
+            {
+                r.SetRootBranch("$/vtccds/trunk");
+                vtccds.Prepare(r);
+            });
+            h.TfsUrl = "https://tfs.codeplex.com:443/tfs/TFS16";
+            h.Run("clone", h.TfsUrl, "$/vtccds/trunk", "Vtccds", "--branches=all", $"--gitignore={gitignoreFile}");
+
+            AssertNewClone("Vtccds", new[] { "refs/heads/master", "refs/remotes/tfs/default" }, commit: "2c047e8c26998bd261ec50716e8574e691efc990");
+
+            AssertNewClone("Vtccds", new[] { "refs/heads/b1", "refs/remotes/tfs/b1" }, commit: "5afc5abed88f15087796e8419aee6c133b8f1786");
+
+            AssertNewClone("Vtccds", new[] { "refs/heads/b1.1", "refs/remotes/tfs/b1.1" }, commit: "cc1cbc9fd62fa7e4ee55a3619858b27c62eb9c00");
+
+            AssertNewClone("Vtccds", new[] { "refs/heads/renameFile", "refs/remotes/tfs/renameFile" }, commit: "cd9aafa3bae84c2b55def73c6dc1cad0dc83dd76");
+
+            AssertNewClone("Vtccds", new[] { "refs/remotes/tfs/branch_from_nowhere" }, commit: "9cb91c60d76d00af182ae9f16da6e6aa77b88a5e");
+
+            AssertNewClone("Vtccds", new[] { "refs/heads/renamed3", "refs/remotes/tfs/renamed3" }, commit: "891e14ba4977835cb1ca06f1ceaa5c7948ea5785");
```

Comment:
Please remove the newlines there, I don't think they help readability. 
Don't get me wrong, I wouldn't have bothered you with these newlines here if this would be the only changes requested, but as we have other stuff left, would be good if you can fix this one, too.

Please also add (in line with the other suggestion further down) either a check that the .gitignore is there with the correct content or as an alternative (as we are not asserting the file contents in this test) a comment ala  `// .gitignore file is included in all  the revs checked here` above these asserts. 

Background: It took me  a while to figure this out, as I wasn't sure if the .gitnore file will be lost when creating a branch. Short answer, it doesn't, but from simply reading the test in isolation this was not clear to me. So leaving some breadcrumbs would help the next reader of the test.

Reply:
I copied WhenCloningFunctionalTestVtccdsWithBranchesRenamingAndGitignore_ThenAllRenamesShouldBeWellHandled, with the newlines, from WhenCloningFunctionalTestVtccdsWithBranchesRenaming_ThenAllRenamesShouldBeWellHandled. I agree that the code looks neater without the newlines, so I removed them from both tests.

My main goal in adding this test was to exercise the complex renaming scenarios in the Vtccds test TFS project with the --gitignore option so that we have coverage both with and without. Therefore, my primary concern was that the resulting git trees are equivalent, and that is what I focused on with the AssertNewClone tests. As you noted, the commit hashes implicitly verify the the inclusion of the .gitignore file with the proper contents in the proper position, so I didn't include a separate assertion for it. 

I hope the comments I added address your concern.

Reply:
Yes, it adresses my concerns. As I said, i figured it out with the hashes, but it took me quite a while. Your newly added comments would have helped. Thanks